### PR TITLE
Implement #154: add client verbosity; make default less verbose

### PR DESCRIPTION
**Pull request by hasLeland**
This PR changes the way the generated CLI will print it's output by default. This change would have the client by default only print the json form of the input and the output, with errors automatically converted to json format. However, the user may tell the client to print more information using the newly added `-verbose` flag.

### CODE REVIEW

File: gengokit/httptransport/templates.go [hunk 1]
```diff
@@ -233,6 +237,7 @@ var (
 	_ = ioutil.NopCloser
 	_ = pb.Register{{.Service.Name}}Server
 	_ = io.Copy
+	Verbose = false
```
Comment (adamryman) [inline gengokit/httptransport/templates.go]: Could this be in its own declaration block with a comment? 

Perhaps a comment could be added to the `_` block and mention it is for compiling even if those packages are not used.

**Review by adamryman — COMMENTED** [review]
Request:

Could you split up the templates like `gengokit/middlewares/templates` is split up? 

Create a `gengokit/httptransport/templates` directory and have 2 files.

Name them so we know which is for the client and which is the server:

Maybe
`server.go` (which becomes transport_http.go)
`client.go` (which becomes client/http/client.go)

Probably deserves it's own pull request, though I think this one will be easier to follow afterwards. After that gets merged in, you can put this one on top of that one.

File: gengokit/httptransport/templates.go (hunk 1 above)
Comment (hasLeland) [reply gengokit/httptransport/templates.go]: That seems like an excellent change, will do!

**Comment by adamryman** [comment]
Looking good! 

I think you are correct that adding this as a global variable feels bad. It is an improvement over sending it through the context though.

Now that the templates are separated some more, and idea came to mind. Why don't we move all the Encoding / Decoding methods that the client uses, to the client package?

That way setting the verbosity could be one of the `ClientOption...` that can be passed in.

The only reason that the methods are in the `transport_http.go` file, is because that is where they existed in the go-kit example.

Also could you move the separating of the templates to a different PR? I want to get that merged in.

**Pull request closed without merging**